File: flutter/lib/domain/ship_navigation/find_middle_points.dart
import 'package:ship_conquest/domain/space/map_position.dart';

List<MapPosition> defineMiddlePoints(List<MapPosition> path, int numPoints) {
  List<MapPosition> extractedPoints = List.empty(growable: true);
  final double step = (path.length - 1).toDouble() / (numPoints - 1).toDouble();
  for(int i = 0; i < numPoints; i++) {
    final int index = (i * step).toInt();
    extractedPoints.add(path[index]);
  }
  return extractedPoints;
}

File: jvm/ship-conquest/src/main/kotlin/com/example/shipconquest/domain/path_finding/findMiddlePoints.kt
package com.example.shipconquest.domain.path_finding

import com.example.shipconquest.domain.Vector2


/**
 * Define middle points in the path to prepare the usage of Bézier curves at equal distance
 * @param path the shortest path to be used
 * @param distance the desired distance between points
 * @return a list with the points of interest
 */
fun defineMiddlePoints(path: List<Vector2>, numPoints: Int): List<Vector2> {
    val extractedPoints = mutableListOf<Vector2>()
    val step = (path.size - 1).toDouble() / (numPoints - 1).toDouble()
    for (i in 0 until numPoints) {
        val index = (i * step).toInt()
        extractedPoints.add(path[index])
    }
    return extractedPoints
}
